src/App.js

```diff
@@ -29,37 +29,108 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [playerTurn, setPlayerTurn] = useState(PLAYER_1);
+  const [won, setWon] = useState(null);
 
   // Wave 2
   // You will need to create a method to change the square
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
+  const updateBoard = markedSquare => {
+    
+    const newBoard = [...squares];
```

Comment:
Great job in making a copy of the squares data!  React can't check for changes within  a mutable data type like arrays which is why we should pass in a copy to the state function to trigger a re-render.